libs/dal/src/repositories/notification-template/notification-template.repository.ts

```diff
@@ -1,25 +1,53 @@
-import { Document, FilterQuery } from 'mongoose';
+import { Document, FilterQuery, ProjectionType } from 'mongoose';
 import { SoftDeleteModel } from 'mongoose-delete';
 import { BaseRepository, Omit } from '../base-repository';
 import { NotificationTemplate } from './notification-template.schema';
 import { NotificationTemplateEntity } from './notification-template.entity';
-import { DalException } from '../../shared';
+import { Cached, DalException, ICacheService, InvalidateCache } from '../../shared';
 
-class PartialIntegrationEntity extends Omit(NotificationTemplateEntity, ['_environmentId', '_organizationId']) {}
+class PartialNotificationTemplateEntity extends Omit(NotificationTemplateEntity, [
+  '_environmentId',
+  '_organizationId',
+]) {}
 
-type EnforceEnvironmentQuery = FilterQuery<PartialIntegrationEntity & Document> &
+type EnforceEnvironmentQuery = FilterQuery<PartialNotificationTemplateEntity & Document> &
   ({ _environmentId: string } | { _organizationId: string });
 
+type EnforceIdentifierQuery = FilterQuery<PartialNotificationTemplateEntity & Document> & { _environmentId: string } & {
```

Comment:
This type for me means more `EnforceEnvironmentQuery` than the previous one. Could we consider in swapping their names as the identification is more accurate with environmentId and organizationId.

Reply:
Makes sense to me, thanks for raising it up!